### PR TITLE
Revised "Contributor Guide" (style, formatting, and file structure)

### PR DESCRIPTION
**Pull request by chanmosq**
**There are no major updates to the page content - they were simply moved to another file, thus the higher lines-updated count.** 

- Moved content of `_index.md` to `overview.md`. (There were no changes in the content). The purpose of this change is so the first page of the section is called "Overview", instead of having the same title as the section title. 
- Shortened all instances of "Open 3D Engine" in page titles to "O3DE" and made sure all pages mentioned "**Open 3D Engine (O3DE)**" in the first paragraph. The purpose of this change is to make long page titles more concise, and apply that change so its consistent with other pages. 
- Other minor style and formatting revisions.

Signed-off-by: Chanelle Mosquera <75444793+chanmosq@users.noreply.github.com>

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/contributing/to-docs/hugo.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 ---
 linkTitle: Hugo
-title: O3DE Docs Contribution with Hugo
+title: O3DE Documentation Contribution with Hugo
 description: Learn to work with Hugo, the static site generator for O3DE documentation
```
Comment (FiniteStateGit) [inline content/docs/contributing/to-docs/hugo.md:4]:
```suggestion
description: Learn to work with Hugo, the static site generator for O3DE documentation.
```

**Review by FiniteStateGit — CHANGES_REQUESTED** [review]
I think that the first two links on the table in `/docs/contributing/_index.md`, could point to `/to-docs/overview` and `/to-code/overview`.

**Comment by chanmosq** [comment]
> I think that the first two links on the table in `/docs/contributing/_index.md` could point to `/to-docs/overview` and `/to-code/overview`.

The table of contents (Section / Description) table in the [Contributing](https://www.o3de.org/docs/contributing/) landing page? They currently point to the **To Code** and **To Docs** sections. Not sure where you mean?

**Review by FiniteStateGit — APPROVED** [review]

**Review by sptramer — APPROVED** [review]
Approved pending the (easy-looking) merge conflict. @chanmosq address that and I can merge.

**Comment by chanmosq** [comment]
@sptramer Resolved merge conflicts!

**Review by sptramer — APPROVED** [review]

**Pull request merged**